preupg/xccdf.py

```diff
@@ -49,94 +49,88 @@ def get_and_print_inplace_risk(verbose, inplace_risk):
-                """
-                if not results[result]:
-                    ret_val = XccdfHelper.get_and_print_inplace_risk(verbose, results[result])
-                    if result == 'fail' and int(ret_val) == -1:
-                        current_val = settings.RESULT_BASED_RETURN_CODES['error']
-                    else:
-                        current_val = settings.RESULT_BASED_RETURN_CODES[result]
-                else:
-                    ret_val = XccdfHelper.get_and_print_inplace_risk(verbose, results[result])
-                    logger_report.debug('Return value from "get_and_print_inplace_risk" is %s' % ret_val)
-                    if result == 'fail' and int(ret_val) == -1:
-                        current_val = settings.RESULT_BASED_RETURN_CODES['error']
-                    elif result in settings.ERROR_RETURN_VALUES and int(ret_val) != -1:
-                        current_val = settings.RESULT_BASED_RETURN_CODES['error']
-                    elif int(ret_val) == -1:
-                        current_val = settings.RESULT_BASED_RETURN_CODES[result]
-                    else:
-                        # EXTREME has to return 2 as FAIL
-                        if ret_val == 2:
-                            current_val = ResultBasedReturnCodes.FAIL
-                        else:
-                            # Needs_action has to return 1 as needs_inspection
-                            current_val = ResultBasedReturnCodes.NEEDS_INSPECTION
-                """
-                if return_val < current_val:
-                    return_val = current_val
-        return return_val
+        report_results = {}
+        profile = target_tree.find(XMLNS + "TestResult")
+        for rule_result in profile.findall(XMLNS + "rule-result"):
+            result_value = rule_result.find(XMLNS + "result").text
+            inplace_risks = XccdfHelper.get_check_import_inplace_risks(
+                rule_result)
+            if result_value not in report_results.keys():
+                report_results[result_value] = []
+            for risk in inplace_risks:
+                if risk not in report_results[result_value]:
+                    report_results[result_value].append(risk)
+        logger_report.debug(report_results)
+        return report_results
+
+    @staticmethod
+    def get_highest_return_code(report_results, verbose):
+        """Return a number based on the modules' results (exit_x)
+        and inplace risks (log_x_risk). From these the one with the highest
+        severity is chosen.
+        """
+        highest_return_code = settings.ResultBasedReturnCodes.PASS
+        for found_result in report_results.keys():
+            if found_result not in settings.RESULT_BASED_RETURN_CODES.keys():
+                sys.stderr.write("Unknown result: %s\n" % found_result)
+                sys.exit(settings.PreupgReturnCodes.INTERNAL_EXCEPTION)
+                continue
```

Comment:
The `continue` doesn't make sense here as the `sys.exit` is called before. It is dead code.